src/viur/core/modules/user.py

```diff
@@ -972,26 +1138,42 @@ def getCurrentUser(self):
 
     def continueAuthenticationFlow(self, caller, userKey):
         session = current.session.get()
-        session["_mayBeUserKey"] = userKey.id_or_name
+        session["possible_user_key"] = userKey.id_or_name
         session["_secondFactorStart"] = utils.utcNow()
         session.markChanged()
 
-        for authProvider, secondFactor in self.validAuthenticationMethods:
-            if isinstance(caller, authProvider):
-                if secondFactor is None:
-                    # We allow sign-in without a second factor
-                    return self.authenticateUser(userKey)
-                # This Auth-Request was issued from this authenticationProvider
-                secondFactorProvider = self.secondFactorProviderByClass(secondFactor)
-                if secondFactorProvider.canHandle(userKey):
-                    # We choose the first second factor provider which claims it can verify that user
-                    return secondFactorProvider.startProcessing(userKey)
-        # Whoops.. This user logged in successfully - but we have no second factor provider willing to confirm it
-        raise errors.NotAcceptable("There are no more authentication methods to try")  # Sorry...
+        second_factor_providers = []
+
+        if not (possible_user := db.Get(userKey)):
+            raise errors.NotFound()
+        for auth_provider, second_factor in self.validAuthenticationMethods:
+            if isinstance(caller, auth_provider):
+                if second_factor is not None:
+                    second_factor_provider_instance = self.secondFactorProviderByClass(second_factor)
+                    if second_factor_provider_instance.can_handle(possible_user):
+                        second_factor_providers.append(second_factor_provider_instance)
+                else:
+                    second_factor_providers.append(None)
+        if len(second_factor_providers) > 1 and None in second_factor_providers:
+            # We have a second factor. So we can get rid of the None
+            second_factor_providers.pop(second_factor_providers.index(None))
+
+        if len(second_factor_providers) == 0:
+            raise errors.NotAcceptable("There are no authentication methods to try")
+        elif len(second_factor_providers) == 1:
+            if second_factor_providers[0] is None:
+                # We allow sign-in without a second factor
+                return self.authenticateUser(userKey)
+            else:
+                # We have only one second factor we don't need the choice template
+                return second_factor_providers[0].start(userKey)
+        else:
+            # We have more than one second factor we need the choice template
+            return self.render.second_factor_choice(tpl="second_factor_choice", second_factors=second_factor_providers)
```

Comment:
```suggestion
        # In case there is more than one second factor, let the user select a method.
        return self.render.second_factor_choice(tpl="second_factor_choice", second_factors=second_factor_providers)
```